Dataset.py

```diff
@@ -157,7 +157,12 @@ def __init__(self, name=None,
     elif isinstance(context_window, int):
       context_window = NumbersDict(broadcast_value=0, numbers_dict={"data": context_window})
     elif isinstance(context_window, dict):
-      context_window = NumbersDict(broadcast_value=0, numbers_dict=context_window)
+      if set(context_window.keys()) == {"left", "right"}:
```

Comment:
This is not a good idea. This is incompatible with the behavior it had before. Before, the keys were the data keys. Now you have a special behavior for this specific case. What e.g. if your data keys are "left" and "right"?

I would suggest to make your case if `isinstance(context_window, (tuple, list))`.

Also, it is even uglier that you have the same strange check below.
To avoid this, directly set `ctx_left` and `ctx_right` here.